src/uthash.h

```diff
@@ -188,79 +190,117 @@ do {
   (head)->hh.tbl->signature = HASH_SIGNATURE;                                    \
 } while(0)
 
-#define HASH_ADD(hh,head,fieldname,keylen_in,add)                                \
-        HASH_ADD_KEYPTR(hh,head,&((add)->fieldname),keylen_in,add)
+#define HASH_SADD(hh,head,fieldname,keylen_in,add,cmpfcn)                        \
+    HASH_SADD_KEYPTR(hh,(head),&((add)->fieldname),(keylen_in),(add),cmpfcn)
+
+#define HASH_SADD_BYHVAL(hh,head,fieldname,keylen_in,hashval,add,cmpfcn)         \
```

Comment:
This should read `..._BY_HVAL` under your new naming convention; but why did you change it at all? Is something wrong with `..._BY_HASH_VALUE`?